src/main/jbake/content/blog/2017/07/opentracing-ejb.adoc

```diff
@@ -0,0 +1,393 @@
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Stateless
+@Interceptors(OpenTracingInterceptor.class)
+public class AccountService {
+    private static final Logger log = Logger.getLogger(AccountService.class.getName());
+
+    public void sendNotification() {
+        log.info("Notifying the account owner about a new order");
+    }
+}
+```
+
+Our `OrderService` is responsible for actually placing the order: it's where the business knowledge
+resides. We'll later look into details at the `InventoryService`, but for now, we need to know that
+this service requires a `SpanContext` to be explicitly called. We can get this context from the `EJBContext`,
+stored under a context data entry that can be retrieved with the constant 
+`io.opentracing.contrib.ejb.OpenTracingInterceptor.SPAN_CONTEXT`.
+
+`src/main/java/io/opentracing/contrib/ejb/example/OrderService.java`:
+```java
+package io.opentracing.contrib.ejb.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.ejb.OpenTracingInterceptor;
+
+import javax.annotation.Resource;
+import javax.ejb.EJBContext;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.interceptor.Interceptors;
+import java.util.logging.Logger;
+
+import static io.opentracing.contrib.ejb.OpenTracingInterceptor.SPAN_CONTEXT;
+
+/**
+ * This is a regular synchronous stateless EJB. It demonstrates how to get the span context for the span it's wrapped
+ * on. This can be used to pass down the call chain, create child spans or add baggage items.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Stateless
+@Interceptors(OpenTracingInterceptor.class)
+public class OrderService {
+    private static final Logger log = Logger.getLogger(OrderService.class.getName());
+
+    @Resource
+    EJBContext ctx;
+
+    @Inject
+    InventoryService inventoryService;
+
+    public void processOrderPlacement() {
+        log.info("Placing order");
+        Object ctxParentSpan = ctx.getContextData().get(SPAN_CONTEXT);
+        if (ctxParentSpan instanceof SpanContext) {
+            inventoryService.changeInventory((SpanContext) ctxParentSpan);
+            return;
+        }
+
+        inventoryService.changeInventory(null);
+    }
+}
+```
+
+Our `InventoryService` is responsible for interfacing with backend systems dealing with inventory control.
+We don't want to hold the parent transaction while interacting with those systems, so, we make this as an
+asynchronous EJB. As such, we cannot rely on the context data from the `EJBContext`.
+
+The OpenTracing EJB integration is able to intecept the method call and detect if there is a span context
+among the parameters, which is the case of our `changeInventory(SpanContext)` method. On such case,
+the following happens behind the scenes:
+
+1. The caller makes a method call, passing the `SpanContext`
+1. The interceptor comes to the scene, create a new child span using the `SpanContext` as the parent
```

Comment:
"The interceptor is invoked, creating a new child..." (possibly activated instead of invoked?)